core/src/stored/backends/crud_storage.cc

```diff
@@ -0,0 +1,362 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+   Copyright (C) 2024-2024 Bareos GmbH & Co. KG
+
+   This program is Free Software; you can redistribute it and/or
+   modify it under the terms of version three of the GNU Affero General Public
+   License as published by the Free Software Foundation and included
+   in the file LICENSE.
+
+   This program is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+   Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301, USA.
+*/
+#include "include/bareos.h"
+#include <fmt/format.h>
+#include "crud_storage.h"
+#include "lib/berrno.h"
+#include "lib/bpipe.h"
+#include "lib/bstringlist.h"
+#include "stored/stored_conf.h"
+#include "stored/stored_globals.h"
+#include <sys/stat.h>
+#include <cctype>
+#if defined(HAVE_WIN32)
+#  include <shlwapi.h>  // for PathIsRelative()
+#endif
+
+namespace {
+constexpr int debug_info = 110;
+constexpr int debug_trace = 130;
+
+bool path_is_relative(const std::string& path)
+{
+#if defined(HAVE_WIN32)
+  return PathIsRelative(path.c_str());
+#else
+  return path[0] != '/';
+#endif
+}
+
+class BPipeHandle {
+  Bpipe* bpipe{nullptr};
+
+ public:
+  BPipeHandle(const char* prog,
+              int wait,
+              const char* mode,
+              const std::unordered_map<std::string, std::string>& env_vars = {})
+      : bpipe(OpenBpipe(prog, wait, mode, true, env_vars))
+  {
+    if (!bpipe) { throw std::system_error(ENOENT, std::generic_category()); }
```

Comment:
I just realized that if OpenBpipe() fails and the exception is thrown the SD would just terminate, because the exception is never caught anywhere.
I guess I have some more work to do :(

Reply:
I hope what I did was an improvement. Could you take another look?

Reply:
Yeah its great.  It might be smart to just make the constructor private so that you can never accidentally "leak" an exception, but this is probably already enough.